epochStart/shardchain/trigger.go

```diff
@@ -181,14 +190,56 @@ func NewEpochStartTrigger(args *ArgsShardEpochStartTrigger) (*trigger, error) {
 		appStatusHandler:            &statusHandler.NilStatusHandler{},
 	}
 
-	newTrigger.headersPool.RegisterHandler(newTrigger.receivedMetaBlock)
+	t.headersPool.RegisterHandler(t.receivedMetaBlock)
 
-	err := newTrigger.saveState(newTrigger.triggerStateKey)
+	err := t.saveState(t.triggerStateKey)
 	if err != nil {
 		return nil, err
 	}
 
-	return newTrigger, nil
+	t.mapMisingMiniblocks = make(map[string]*block.MiniBlock)
+	go t.requestMissingMiniblocks()
+
+	return t, nil
+}
+
+func (t *trigger) requestMissingMiniblocks() {
+	for {
+		time.Sleep(sleepTime)
+
+		t.mutMissingMiniblocks.RLock()
+		if len(t.mapMisingMiniblocks) == 0 {
+			t.mutMissingMiniblocks.RUnlock()
+			continue
+		}
+
+		missingMiniblocks := make([][]byte, len(t.mapMisingMiniblocks))
+		for hash := range t.mapMisingMiniblocks {
```

Comment:
missing are only those which are nil in the map.

Reply:
Refactored